edge-util/src/Microsoft.Azure.Devices.Edge.Util/CertificateHelper.cs

```diff
@@ -373,67 +379,103 @@ internal static X509Certificate2 ParseManifestTrustedBundleCerts(string manifest
+                    }
+
+                    if (certObject is RsaPrivateCrtKeyParameters rsaParameters)
+                    {
+                        keyParams = rsaParameters;
+                    }
+                    else if (certObject is ECPrivateKeyParameters ecParameters)
+                    {
+                        isEcKey = true;
+                        keyParams = ecParameters;
+                    }
+
+                    certObject = pemReader.ReadObject();
                 }
 
-                if (certObject is RsaPrivateCrtKeyParameters rsaParameters)
+                if (keyParams == null)
                 {
-                    keyParams = rsaParameters;
+                    throw new InvalidOperationException("Private key is required");
                 }
-                else if (certObject is ECPrivateKeyParameters ecParameters)
+
+                store.SetKeyEntry("Edge", new AsymmetricKeyEntry(keyParams), chain.ToArray());
+                using (var p12File = new MemoryStream())
                 {
-                    keyParams = ecParameters;
-                }
+                    store.Save(p12File, new char[] { }, new SecureRandom());
 
-                certObject = pemReader.ReadObject();
-            }
+                    var cert = new X509Certificate2(p12File.ToArray());
 
-            if (keyParams == null)
-            {
-                throw new InvalidOperationException("Private key is required");
-            }
+                    // On Windows, from time to time the private key cannot be accessed and kestrel fails accepting connections
+                    // without the private key. Testing the private key here and if it fails, retry the entire cert import.
+                    // The second try usually works.
+                    try
+                    {
+                        if (cert.HasPrivateKey)
+                        {
+                            if (isEcKey)
+                            {
+                                _ = cert.GetECDsaPrivateKey();
+                            }
+                            else
+                            {
+                                _ = cert.GetRSAPrivateKey();
+                            }
 
-            store.SetKeyEntry("Edge", new AsymmetricKeyEntry(keyParams), chain.ToArray());
-            using (var p12File = new MemoryStream())
-            {
-                store.Save(p12File, new char[] { }, new SecureRandom());
+                            return (cert, certsChain);
+                        }
+                    }
+                    catch
+                    {
+                        // swallow
+                    }
 
-                var cert = new X509Certificate2(p12File.ToArray());
-                return (cert, certsChain);
+                    logger?.LogWarning("Error importing certificate, retrying");
+                    Thread.Sleep(TimeSpan.FromSeconds(1)); // Do not spam the log
```

Comment:
I am fine with it... Why do we need to sleep here? Is this related to some timing issue?